BoltsTest/src/bolts/TaskTest.java

```diff
@@ -268,7 +270,251 @@ public void testWhenAllNoTasks() {
+          public Void call() throws Exception {
+            Thread.sleep(50);
+            throw new CancellationException();
+          }
+        });
+        tasks.addAll(launchTasksWithRandomCompletions(10));
+        tasks.add(firstToCompleteCancelled);
+        return Task.whenAny(tasks).continueWith(new Continuation<Task<?>, Object>() {
+          @Override
+          public Object then(Task<Task<?>> task) throws Exception {
+            assertTrue(task.isCompleted());
+            assertFalse(task.isFaulted());
+            assertFalse(task.isCancelled());
+            assertEquals(task.getResult(), firstToCompleteCancelled);
+            assertTrue(task.getResult().isCompleted());
+            assertTrue(task.getResult().isCancelled());
+            assertFalse(task.getResult().isFaulted());
+            return null;
+          }
+        });
+      }
+    });
+  }
+
+  private Collection<Task<Void>> launchTasksWithRandomCompletions(int numberOfTasksToLaunch ) {
+    final Exception error = new RuntimeException("This task failed.");
+    final ArrayList<Task<Void>> tasks = new ArrayList<>();
+    for (int i=0; i < numberOfTasksToLaunch; i++) {
+      Task<Void> task = Task.callInBackground(new Callable<Void>() {
+        @Override
+        public Void call() throws Exception {
+          Thread.sleep((long) (500 + (Math.random() * 100)));
+          double rand = Math.random();
+          if (rand >= 0.7) {
+            throw error;
+          } else if (rand >= 0.4) {
+          throw new CancellationException();
+          }
+          return null;
+        }
+      });
+      tasks.add(task);
+    }
+    return tasks;
+  }
+
+  public void testWhenAny() {
+    runTaskTest(new Callable<Task<?>>() {
+        @Override
+        public Task<?> call() throws Exception {
+            final ArrayList<Task<?>> tasks = new ArrayList<Task<?>>();
+            final Task<Integer> tLongRunningOperation = Task.callInBackground(new Callable<Integer>() {
+                @Override
+                public Integer call() throws Exception {
+                    Thread.sleep(500);
+                    return 2;
+                }
+            });
+            final Task<Void> tTimeout = Task.callInBackground(new Callable<Void>() {
+                @Override
+                public Void call() throws Exception {
+                    Thread.sleep(1500);
+                    return null;
+                }
+            });
+            tasks.add(tLongRunningOperation);
+            tasks.add(tTimeout);
+            return Task.whenAny(tasks).continueWith(new Continuation<Task<?>, Object>() {
+                @Override
+                public Object then(Task<Task<?>> task) throws Exception {
+                    assertEquals(task.getResult(), tLongRunningOperation);
+                    assertEquals(Integer.valueOf(2), tLongRunningOperation.getResult());
+                    return null;
+                }
+            });
+        }
+    });
+  }
```

Comment:
nit: indentation looks off here

This test also seems a bit redundant with the additions to the previous tests.